envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -126,34 +127,43 @@ fun Value.toIncomingEndpoint(): IncomingEndpoint {
     }
 }
 
-private fun Value?.toIncomingTimeoutPolicy(): Incoming.TimeoutPolicy {
-    val idleTimeout: Duration? = this?.field("idleTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
-    val responseTimeout: Duration? = this?.field("responseTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
-
-    val connectionIdleTimeout: Duration? = this?.field("connectionIdleTimeout")?.stringValue
-        ?.takeIf { it.isNotBlank() }
-        ?.let { Durations.parse(it) }
+fun Value?.toIncomingTimeoutPolicy(): Incoming.TimeoutPolicy {
```

Comment:
It seems to me that the API in this module is somewhat inconsistent. `toIncomingTimeoutPolicy` is public, while `toOutgoingTimeoutPolicy` - I don't see the reasons yet for making them public, but at least they should have the same visibility.